sway/input/seat.c

```diff
@@ -823,6 +821,9 @@ void seat_configure_device(struct sway_seat *seat,
 			seat_configure_tablet_pad(seat, seat_device);
 			break;
 	}
+	if (wlr_cursor_is_input_device_attached(seat->cursor->cursor, input_device->wlr_device)) {
```

Comment:
Why isn't `if (input_device->wlr_device->type != WLR_INPUT_DEVICE_KEYBOARD && input_device->wlr_device->type != WLR_INPUT_DEVICE_SWITCH)` enough?

Reply:
That doesn't guarantee that the corresponding device is associated with the seat's cursor

Reply:
`seat_configure_device` is not called if the device doesn't belong to the seat.

Reply:
`wlr_cursor_is_input_device_attached` should take that into consideration

Reply:
I don't get it.

The whole function is not called if the device isn't attached to the seat, so there's no point in checking that here.

Reply:
All the devices are attached to the seat as a prerequisite. `wlr_cursor_is_input_device_attached` checks if the specified device controls the seat's cursor. Then function uses the wlr_cursor as the "object" not the seat itself.

Reply:
There's no need to check that.

For instance, let's say a pointer gets added to the seat. `seat_configure_pointer` gets called. It itself calls `wlr_cursor_attach_input_device`. Then `wlr_cursor_is_input_device_attached` will always return true.

Reply:
But `seat_configure_device` runs for all devices, so non-pointing devices have to be ruled out or only pointing devices should be included. To prevent code duplicates in the device specific configure functions of pointing devices, this will emulate cursor activity to arm the hide timeouts. 

I chose this method since it is more easily maintained in such a case where a new kind of pointing device is added.

Reply:
I'd prefer not to add a new wlroots function just because of how the Sway code is architectured.

Reply:
Well the functionality was already in wlroots. It just wasn't publicly accessible.